src/components/Cardlist/cardlist.tsx

```diff
@@ -0,0 +1,72 @@
+import { useEffect, useRef, useState } from "react";
+import Cards from "../Cards/cards";
+import { ArticlesData, articles } from "../../main";
+import "./cardlist.css";
+import "./search.css";
+
+const Cardlist = (): JSX.Element => {
+    const [keyword, setKeyword] = useState("");
+    const [filteredArticles, setFilteredArticles] = useState(articles);
+    const inputTextRef = useRef<HTMLInputElement | null>(null);
+    
+    const handleInputChange = () => {
+        inputTextRef.current ? setKeyword(inputTextRef.current.value) : setKeyword("");
+    }
+    
+    const handleSubmit = (event: React.FormEvent) => {
+        event.preventDefault();
+        handleInputChange();
+    }
+
+    const handleReset = () => {
+        if (inputTextRef.current) {
+            inputTextRef.current.value = "";
+            setKeyword("");
+        }
+    }
+
+    const search = (articles: ArticlesData[], keyword: string): ArticlesData[] => {
+        if (!keyword) {
+            return articles;
+        } 
+        
+        return articles.filter(article => (
+            article.title.toLowerCase().includes(keyword.toLowerCase())))
+    }
+
+    useEffect( () => {
```

Comment:
Also obsolete. You would only need useEffect if the keyworld would somehow persist after page reload.